components/job-orchestration/job_orchestration/scheduler/query/query_scheduler.py

```diff
@@ -495,16 +594,15 @@ async def check_job_status_and_update_db(db_conn_pool, results_cache_uri):
             id for id, job in active_jobs.items() if InternalJobState.RUNNING == job.state
         ]:
             job = active_jobs[job_id]
-            is_reducer_job = job.reducer_handler_msg_queues is not None
-
             try:
                 returned_results = try_getting_task_result(job.current_sub_job_async_task_result)
             except Exception as e:
                 logger.error(f"Job `{job_id}` failed: {e}.")
                 # Clean up
-                if is_reducer_job:
-                    msg = ReducerHandlerMessage(ReducerHandlerMessageType.FAILURE)
-                    await job.reducer_handler_msg_queues.put_to_handler(msg)
+                if QueryJobType.SEARCH == job.type():
+                    if job.reducer_handler_msg_queues is not None:
```

Comment:
Can we join these?

Reply:
I didn't join them because I feel in this way it's more obvious that we only check the second condition if the first condition is met.

I can join them if you prefer having them in one line of code

Reply:
Fair enough